### PR TITLE
senMail

### PR DESCRIPTION
**Pull request by Le-Caignec**

### CODE REVIEW

File: src/features/sendEmail/SendEmail.tsx [hunk 1]
```diff
@@ -31,9 +38,36 @@ export default function SendMessageTo() {
     setCharactersRemaining(500 - inputValue.length);
   };
 
+  const sendEmailHandle = () => {
+    if (!address) return;
+    sendEmail({
+      emailSubject: messageObject,
+      emailContent: value,
+      protectedData: address,
```
Comment (PierreJeanjacquot) [inline src/features/sendEmail/SendEmail.tsx:46]: :bug: the address used here is the user address, it must be a protected data address

File: src/features/sendEmail/SendEmail.tsx [hunk 2]
```diff
@@ -31,9 +38,36 @@ export default function SendMessageTo() {
...
+    if (reason === 'clickaway') {
+      return;
+    }
+    setOpen(false);
+  };
+
   return (
     <Box sx={{ m: 10, mx: 20 }}>
-      <h2>Send Message to {receiverAddress}</h2>
+      <h2>Send Mail to {receiverAddress}</h2>
```
Comment (PierreJeanjacquot) [inline src/features/sendEmail/SendEmail.tsx:70]:
```suggestion
      <h2>Send EMail to {receiverAddress}</h2>
```

File: src/features/sendEmail/SendEmail.tsx [hunk 3]
```diff
@@ -58,10 +92,25 @@ export default function SendMessageTo() {
...
+        autoHideDuration={6000}
+        onClose={handleClose}
+        anchorOrigin={{ horizontal: 'right', vertical: 'top' }}
+      >
+        <Alert
+          onClose={handleClose}
+          severity={success ? 'success' : 'error'}
+          sx={{ width: '100%' }}
+        >
+          {success ? 'The email has been sent!' : 'Failed to send the Email!'}
```
Comment (PierreJeanjacquot) [inline src/features/sendEmail/SendEmail.tsx:111]: consistency

```suggestion
          {success ? 'The email has been sent!' : 'Failed to send the email!'}
```

File: package.json [hunk 4]
```diff
@@ -13,8 +13,8 @@
   "dependencies": {
     "@emotion/react": "^11.10.6",
     "@emotion/styled": "^11.10.6",
-    "@iexec/dataprotector": "^0.0.0-nightly-8395225223034e4980135f6bf58f0a370721b496",
-    "@iexec/web3mail": "^0.0.0-nightly-cb75c44ed80f3698dd7923d62ccaf609a455305d",
+    "@iexec/dataprotector": "^0.1.0",
+    "@iexec/web3mail": "^0.0.0-nightly-e93b6b6eaa87abca06f8a829bb5b430423b10df8",
```
Comment (PierreJeanjacquot) [inline package.json:17]: use fixed version when the version does not respect semver

```suggestion
    "@iexec/web3mail": "0.0.0-nightly-e93b6b6eaa87abca06f8a829bb5b430423b10df8",
```

File: src/features/emailDapp/EmailDapp.tsx [hunk 5]
```diff
@@ -61,11 +61,11 @@ export default function EmailDapp() {
           color="secondary"
           onClick={() =>
             navigate(
-              `./sendMessageTo/${params.row.owner}/${params.row.protectedDataAddress}`
+              `./${params.row.owner}/${params.row.protectedDataAddress}`
             )
           }
         >
-          Send Message
+          Send Mail
```
Comment (PierreJeanjacquot) [inline src/features/emailDapp/EmailDapp.tsx:68]:
```suggestion
          Send Email
```

File: src/features/sendEmail/SendEmail.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  const [value, setValue] = useState('');
+  //limited to 4096 by the SMS
+  const [charactersRemainingMessage, setCharactersRemainingMessage] =
+    useState(4096);
+
+  //for name et dataType
+  const [messageSubject, setMessageSubject] = useState('');
+  //limited to 78 by the SMS
+  const [charactersRemainingSubject, setCharactersRemainingSubject] =
+    useState(78);
```
Comment (PierreJeanjacquot) [inline src/features/sendEmail/SendEmail.tsx:31]: do not add to the state something that is derived state

```ts
const charactersRemainingSubject = 78 - messageSubject.length;
```

File: src/features/sendEmail/SendEmail.tsx [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+  const { receiverAddress, protectedDataAddress } = useParams();
+
+  //RTK Mutation hook
+  const [sendEmail, result] = useSendEmailMutation();
+
+  //for textarea
+  const [value, setValue] = useState('');
+  //limited to 4096 by the SMS
+  const [charactersRemainingMessage, setCharactersRemainingMessage] =
+    useState(4096);
```
Comment (PierreJeanjacquot) [inline src/features/sendEmail/SendEmail.tsx:25]: do not add to the state something that is derived state

```ts
const charactersRemainingSubject = 4096 - value.length;
```

File: src/features/sendEmail/SendEmail.tsx [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+import { useSendEmailMutation } from '../../app/appSlice';
+
+export default function SendEmail() {
+  const { receiverAddress, protectedDataAddress } = useParams();
+
+  //RTK Mutation hook
+  const [sendEmail, result] = useSendEmailMutation();
+
+  //for textarea
+  const [value, setValue] = useState('');
```
Comment (PierreJeanjacquot) [inline src/features/sendEmail/SendEmail.tsx:22]: rename `value` to something more meaningful (ex: `message`)

File: src/features/sendEmail/SendEmail.tsx [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+  const handleMessageSubjectChange = (event: any) => {
+    const inputValue = event.target.value;
+    setMessageSubject(inputValue);
+    setCharactersRemainingSubject(78 - inputValue.length);
+  };
+
+  const handleChange = (event: any) => {
+    const inputValue = event.target.value;
+    setValue(inputValue);
+    setCharactersRemainingMessage(4096 - inputValue.length);
```
Comment (PierreJeanjacquot) [inline src/features/sendEmail/SendEmail.tsx:43]:
```suggestion
```

File: src/features/sendEmail/SendEmail.tsx [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+  const [messageSubject, setMessageSubject] = useState('');
+  //limited to 78 by the SMS
+  const [charactersRemainingSubject, setCharactersRemainingSubject] =
+    useState(78);
+
+  //handle functions
+  const handleMessageSubjectChange = (event: any) => {
+    const inputValue = event.target.value;
+    setMessageSubject(inputValue);
+    setCharactersRemainingSubject(78 - inputValue.length);
```
Comment (PierreJeanjacquot) [inline src/features/sendEmail/SendEmail.tsx:37]:
```suggestion
```

File: src/features/sendEmail/SendEmail.tsx [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+    sendEmail({
+      emailSubject: messageSubject,
+      emailContent: value,
+      protectedData: protectedDataAddress,
+    });
+  };
+
+  //snackbar notification
+  const [open, setOpen] = useState(false);
+  const [success, setSuccess] = useState(false);
```
Comment (PierreJeanjacquot) [inline src/features/sendEmail/SendEmail.tsx:57]: create a ticket in the backlog to add a proper notification system in the whole app

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]
see comments

File: src/features/sendEmail/SendEmail.tsx (hunk 8 above)
Comment (Le-Caignec) [reply src/features/sendEmail/SendEmail.tsx]: Commit: f2d0e679696c52fad4cf991ee9e7a16eb380884b

File: src/features/sendEmail/SendEmail.tsx (hunk 6 above)
Comment (Le-Caignec) [reply src/features/sendEmail/SendEmail.tsx]: Commit: 44edd3cf75d5f0fe9cdeb3dae87fa32d41a0a8b1

File: src/features/sendEmail/SendEmail.tsx (hunk 7 above)
Comment (Le-Caignec) [reply src/features/sendEmail/SendEmail.tsx]: Commit: 44edd3cf75d5f0fe9cdeb3dae87fa32d41a0a8b1

**Pull request merged**